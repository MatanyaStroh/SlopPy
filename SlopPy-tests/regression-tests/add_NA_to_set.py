# try to add an NA object to a set in various ways;
# all attempts should get rejected

x = 1 / 0
assert type(x) is NA

assert set([x]) == set()
assert set([x,x]) == set()
assert set([x,x,x]) == set()
assert set([1,2,x,3,4]) == set([1,2,3,4])
assert set([1,2,x,3,4,x]) == set([1,2,3,4])
assert set([x,1,2,x,3,4,x]) == set([1,2,3,4])
assert set([x,x,x,x,x,1,x]) == set([1])

s = set()
assert len(s) == 0
s.add(x)
assert len(s) == 0
s.update(set([x]))
assert len(s) == 0

