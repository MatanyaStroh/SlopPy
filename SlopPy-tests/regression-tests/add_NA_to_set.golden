{NA <type 'exceptions.ZeroDivisionError'> 'integer division or modulo by zero'
 Traceback (most recent call last):
  File "regression-tests/add_NA_to_set.py", line 4, in <module>
    x = 1 / 0
}
{BUILD_LIST(* NA *) | regression-tests/add_NA_to_set.py <module>:7}
{BUILD_LIST(* NA *) | regression-tests/add_NA_to_set.py <module>:8}
{BUILD_LIST(* NA *) | regression-tests/add_NA_to_set.py <module>:9}
{BUILD_LIST(* NA *) | regression-tests/add_NA_to_set.py <module>:10}
{BUILD_LIST(* NA *) | regression-tests/add_NA_to_set.py <module>:11}
{BUILD_LIST(* NA *) | regression-tests/add_NA_to_set.py <module>:12}
{BUILD_LIST(* NA *) | regression-tests/add_NA_to_set.py <module>:13}
{set.add(NA) | regression-tests/add_NA_to_set.py <module>:17}
{BUILD_LIST(* NA *) | regression-tests/add_NA_to_set.py <module>:19}
